stroom-explorer/stroom-explorer-impl-db/src/main/java/stroom/explorer/impl/db/ExplorerFavDaoImpl.java

```diff
@@ -22,37 +28,24 @@ public class ExplorerFavDaoImpl implements ExplorerFavDao {
 
     @Override
     public void createFavouriteForUser(final DocRef docRef, final String userId) {
-        final int explorerNodeId = getExplorerNodeId(docRef);
-        final int count = JooqUtil.contextResult(explorerDbConnProvider, context -> context
-                .insertInto(EXPLORER_FAVOURITE,
-                        EXPLORER_FAVOURITE.EXPLORER_NODE_ID,
-                        EXPLORER_FAVOURITE.USER_UUID,
-                        EXPLORER_FAVOURITE.CREATE_TIME_MS)
-                .values(explorerNodeId,
-                        userId,
-                        Instant.now().toEpochMilli())
-                .onDuplicateKeyIgnore()
-                .execute());
-
-        if (count != 1) {
-            throw new RuntimeException("Failed to create an EXPLORER_FAVOURITE entry for doc: " +
-                    docRef.getUuid() + ", user: " + userId);
-        }
+        final Integer explorerNodeId = JooqUtil.contextResult(explorerDbConnProvider, context ->
+                getExplorerNodeId(context, docRef).fetchAny(EXPLORER_NODE.ID));
+        Objects.requireNonNull(explorerNodeId);
+        final ExplorerFavouriteRecord record = new ExplorerFavouriteRecord(
+                null,
+                explorerNodeId,
+                userId,
+                Instant.now().toEpochMilli());
+        JooqUtil.tryCreate(explorerDbConnProvider, record, EXPLORER_FAVOURITE.ID);
```

Comment:
Appreciate this is somewhat lacking in javadoc, but this needs to be
```java
JooqUtil.tryCreate(explorerDbConnProvider, record, EXPLORER_FAVOURITE.EXPLORER_NODE_ID, EXPLORER_FAVOURITE.USER_UUID);
```
This is so it can use these two fields that make up the natural unique key to see if the record already exists.